### PR TITLE
View.dispatch_request doesn't take args, raises error about that instead of not implemented error

### PR DESCRIPTION
**Pull request by sherrycherish**
the dispatch_request() does not have any parameters.

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
when taking a look at the code, that is not at all what is intended

File: flask/views.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def as_view(cls, name, *class_args, **class_kwargs):
         """
         def view(*args, **kwargs):
             self = view.view_class(*class_args, **class_kwargs)
-            return self.dispatch_request(*args, **kwargs)
+            return self.dispatch_request()
```
Comment (RonnyPfannschmidt) [inline flask/views.py]: is is explicitly documented that dispatch functions are called with all url rule arguments,

so the fix would be to have the default implementation accept them in order to improve the error message

removing the **needed** and **documented** arguments is certainly wrong

**Comment by sherrycherish** [comment]
why not change the function to 'dispatch_request(*args, **kwargs)' when it is defined initially.

**Comment by davidism** [comment]
I don't understand what problem this fixes.  `dispatch_request` needs to receive the view arguments just like the standard view would, and you're meant to override it in your subclass to take the required parameters.  Would you show an example demonstrating the problem you're fixing?

**Comment by davidism** [comment]
Oh, never mind, you're getting the wrong error message, should be getting the `NotImplementedError`.  The fix here would be to add the `*args, **kwargs` to `dispatch_request` signature.  Adding a more descriptive message to the error would be nice too.

File: flask/config.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def __init__(self, name, get_converter=None):
         self.__name__ = name
         self.get_converter = get_converter
 
-    def __get__(self, obj, type=None):
+    def __get__(self, obj):
```
Comment (RonnyPfannschmidt) [inline flask/config.py]: why that change? i roughly recall the second parameter is part of the descriptor protocol

**Comment by ThiefMaster** [comment]
General advice, you can amend/squash commits and force-push. While we can do it when merging a PR it's cleaner to already amend things if they are not worth separate commits. I think having just this change would be worth merging but not the other (breaking/invalid) change.

```diff
-    def dispatch_request(self):
+    def dispatch_request(self, *args, **kwargs):
```

**Comment by sherrycherish** [comment]
fine

**Pull request closed without merging**